msal/src/main/java/com/microsoft/identity/client/GlobalSettings.java

```diff
@@ -0,0 +1,194 @@
+                listener.onError(new MsalClientException(GLOBAL_INIT_AFTER_PCA_ERROR_CODE,
+                        GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE));
+                return;
+            }
+
+            setGlobalConfiguration(
+                    initializeGlobalConfiguration(context, configFileResourceId),
+                    listener
+            );
+        }
+    }
+
+    /**
+     * Load the global configuration file using the configuration file and a listener.
+     *
+     * @param configFile Configuration file.
+     * @param listener Handles success and error messages.
+     */
+    @WorkerThread
+    public static void loadGlobalConfigurationFile(@NonNull final File configFile,
+                                                   @NonNull final GlobalSettingsListener listener) {
+        synchronized (mGlobalSettingsSingleton.mGlobalSettingsLock) {
+            if (mGlobalSettingsSingleton.mGlobalSettingsInitialized) {
+                listener.onError(new MsalClientException(GLOBAL_ALREADY_INITIALIZED_ERROR_CODE,
+                        GLOBAL_ALREADY_INITIALIZED_ERROR_MESSAGE));
+                return;
+            }
+
+            if (mGlobalSettingsSingleton.mPCAMergeAttempted) {
+                listener.onError(new MsalClientException(GLOBAL_INIT_AFTER_PCA_ERROR_CODE,
+                        GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE));
+                return;
+            }
+
+            setGlobalConfiguration(
+                    initializeGlobalConfiguration(configFile),
+                    listener
+            );
+        }
+    }
+
+    @WorkerThread
+    private static void setGlobalConfiguration(@NonNull final GlobalSettingsConfiguration globalConfiguration,
+                                               @NonNull final GlobalSettingsListener listener) {
+        mGlobalSettingsSingleton.mGlobalSettingsConfiguration = globalConfiguration;
+        mGlobalSettingsSingleton.mGlobalSettingsInitialized = true;
+        listener.onSuccess("Global configuration initialized.");
+    }
+
+    @WorkerThread
+    protected PublicClientApplicationConfiguration mergeConfigurationWithGlobal(final @NonNull PublicClientApplicationConfiguration developerConfig) {
+        synchronized (mGlobalSettingsSingleton.mGlobalSettingsLock) {
+            mPCAMergeAttempted = true;
+
+            if (!mGlobalSettingsSingleton.mGlobalSettingsInitialized) {
+                Logger.warn(TAG + "mergeConfigurationWithGlobal",
+                        GlobalSettings.NO_GLOBAL_SETTINGS_WARNING);
+                return developerConfig;
+            }
+
+            developerConfig.mergeGlobalConfiguration(mGlobalSettingsSingleton.mGlobalSettingsConfiguration);
+            developerConfig.validateConfiguration();
+            return developerConfig;
+        }
+    }
+
+    protected static GlobalSettings getInstance() {
+        return mGlobalSettingsSingleton;
+    }
+
+    protected static void resetInstance() {
```

Comment:
Why do we need this? Javadoc?

Reply:
This is a method i added to reset the static singleton instance during testing

Reply:
And yes i'll add javadoc, good catch